lib/puppet/type/notify.rb

```diff
@@ -6,7 +6,7 @@ module Puppet
   Type.newtype(:notify) do
     @doc = "Sends an arbitrary message to the agent run-time log."
 
-    newproperty(:message) do
+    newproperty(:message, :idempotent => false) do
```

Comment:
This has been debated both ways, @adrienthebo had liked this methodology I think. Allowing someone to mark something via the type interface does have the benefit of re-usability, and in some ways its neater because of that, but we don't know who needs it yet TBH, so this could be YAGNI. One could argue things in the wild might have known idempotency issues, but a flag to hide that might not be what is desired I guess.

There has been some debating of also filtering exec, but I'm unsure if the same implementation applies in this case, since we wouldn't filter because its inherently non-idempotent but for other boring reasons I can get into if you're curious.